ts/Core/Series/Series.ts

```diff
@@ -474,6 +474,19 @@ class Series {
         // point.
         chart.orderSeries(this.insert(chartSeries));
 
+        // If this is the first time we're writing back to options,
+        // and allowMutatingData enabled create
+        // a copy so that we don't mutate the source array. (#4259)
+        if (options.data &&
+            !options.copyOfOriginalData &&
+            !chart.options.chart.allowMutatingData
+        ) {
+            options.data = chart.options.chart.cloningMethod ?
+                chart.options.chart.cloningMethod(options.data) :
+                JSON.parse(JSON.stringify(options.data));
+            options.copyOfOriginalData = true;
```

Comment:
Why do we need this flag?
If we do need this, then I think it shouldn't be stored on `options`. Maybe on a `series`? But if it's on a series, then again - name `series.copyOfOriginalData` is not clear for me.


What about `series.isDataMutated` or `series.isDataHardCoppied?


And, this goes only one way - we can make a hard copy and set `series.isDataMutated = true`, but we don't set it back to `false`.
Use-case: if I do
```js
chart.update({
    chart: {
        allowMutatingData: true
    }
});
```
and then I do `series.setData(data);`, then the object reference will exist, but `series.isDataMutated` will be still `true`, when it should be `false`.